pkg/host-disk/host-disk.go

```diff
@@ -156,49 +156,75 @@ func (hdc *DiskImgCreator) setlessPVCSpaceToleration(toleration int) {
 
 func (hdc DiskImgCreator) Create(vmi *v1.VirtualMachineInstance) error {
 	for _, volume := range vmi.Spec.Volumes {
-		if hostDisk := volume.VolumeSource.HostDisk; hostDisk != nil && hostDisk.Type == v1.HostDiskExistsOrCreate && hostDisk.Path != "" {
-			diskPath := GetMountedHostDiskPath(volume.Name, hostDisk.Path)
-			diskDir := GetMountedHostDiskDir(volume.Name)
-			if _, err := os.Stat(diskPath); os.IsNotExist(err) {
-				availableSize, err := hdc.dirBytesAvailableFunc(diskDir)
-				if err != nil {
-					return err
-				}
-				requestedSize, _ := hostDisk.Capacity.AsInt64()
-				diskSize := requestedSize
-				if uint64(requestedSize) > availableSize {
-					// Some storage provisioners provision less space than requested, due to filesystem overhead etc.
-					// We tolerate some difference in requested and available capacity up to some degree.
-					// This can be configured with the "pvc-tolerate-less-space-up-to-percent" parameter in the kubevirt-config ConfigMap.
-					// It is provided as argument to virt-launcher.
-					toleratedSize := requestedSize * (100 - int64(hdc.lessPVCSpaceToleration)) / 100
-					if uint64(toleratedSize) > availableSize {
-						return fmt.Errorf("unable to create %s, not enough space, demanded size %d B is bigger than available space %d B, also after taking %v %% toleration into account",
-							hostDisk.Path, uint64(requestedSize), availableSize, hdc.lessPVCSpaceToleration)
-					}
-					diskSize = int64(availableSize)
-
-					msg := fmt.Sprintf("PV size too small: expected %v B, found %v B. Using it anyway, it is within %v %% toleration", requestedSize, availableSize, hdc.lessPVCSpaceToleration)
-					log.Log.Info(msg)
-					err = hdc.notifier.SendK8sEvent(vmi, EventTypeToleratedSmallPV, EventReasonToleratedSmallPV, msg)
-					if err != nil {
-						log.Log.Reason(err).Warningf("Couldn't send k8s event for tolerated PV size: %v", err)
-					}
-				}
-				err = createSparseRaw(diskPath, int64(diskSize))
-				if err != nil {
-					log.Log.Reason(err).Errorf("Couldn't create a sparse raw file for disk path: %s, error: %v", diskPath, err)
-					return err
-				}
-			} else if err != nil {
-				return err
-			}
-			// Change file ownership to the qemu user.
-			if err := ephemeraldiskutils.DefaultOwnershipManager.SetFileOwnership(diskPath); err != nil {
-				log.Log.Reason(err).Errorf("Couldn't set Ownership on %s: %v", diskPath, err)
+		if hostDisk := volume.VolumeSource.HostDisk; shouldMountHostDisk(hostDisk) {
+			if err := mountHostDiskAndSetOwnership(vmi, hdc, volume.Name, hostDisk); err != nil {
```

Comment:
Since we're now passing the `hdc DiskImgCreator` as an argument repeatedly, we may want to pass a reference to it, so it's not copied every time